open_samus_returns_rando/files/levels/s110_surfaceb.lua

```diff
@@ -555,7 +551,11 @@ function s110_surfaceb.OnSubAreaChange(_ARG_0_, _ARG_1_, _ARG_2_, _ARG_3_, _ARG_
   Game.SetScenarioItemEnabledByName("ray01", false)
   Game.SetSceneGroupEnabledByName("sg_debris02", false)
   Game.SetSceneGroupEnabledByName("sg_debris03", false)
-  if _ARG_2_ == "collision_camera_021" and _ARG_0_ == "collision_camera_021" and _ARG_3_ == "RidleyCombat" and Game.GetPlayer().INVENTORY:GetItemAmount("ITEM_METROID_COUNT") == 0 then
+  if _ARG_0_ == "collision_camera_017" and _ARG_2_ == "collision_camera_000" and Blackboard.GetProp("PLAYER_INVENTORY", "ITEM_ADN") < 39 then
```

Comment:
Do you know when the Blackboard property get's updated? Normally we ask the inventory (which is also the thing we update) by using `RandomizerPowerup.GetItemAmount(item_id)` which is just shorthand for `Game.GetItemAmount(Game.GetPlayerName(), item_id)`

Might be synced by the game at the same time but I don't know.

Reply:
I'm not sure when it updates, but this was the only way I could think of to get this to work. Could easily test this by starting with 38 dna and then picking up a 39th dna in the section before this and see how it behaves

Reply:
I really prefer to be consistent with the usage. Makes it more readable. :p
But I guess the blackboard property get's updated at the same time.

Wouldn't the easiest way have been to just replace `ITEM_METROID_COUNT` with the `ITEM_ADN`? The line already contained a way on how to obtain the amount of an item.

Reply:
I tried variations of that and also checking for other items and no matter what I did, the condition would never be met and thus never initiate that function. I do not understand why. This was the only method that worked for me.

Reply:
Worked for me with `Game.GetItemAmount(Game.GetPlayerName(), "ITEM_ADN")`